tests/test_ocr.py

```diff
@@ -185,6 +185,38 @@ def test_char_whitelist():
         char_whitelist="0123456789")
 
 
+@requires_tesseract
+@pytest.mark.parametrize("corrections,expected", [
+    # pylint:disable=bad-whitespace
+    # Default ocr output:
+    (None,                                           'OO'),
+    # Corrections string must match entire word:
+    ({'O': '0'},                                     'OO'),
+    ({'OO': '00'},                                   '00'),
+    # Strings are case-sensitive, and they aren't regexes:
+    ({'oo': '00', '[oO]': '0'},                      'OO'),
+    # Regexes do match anywhere:
+    ({re.compile('[oO]'): '0'},                      '00'),
+    # Make sure it tries all the patterns:
+    ({'AA': 'BB', 'OO': '00'},                       '00'),
+    ({re.compile('^O'): '1', re.compile('O$'): '2'}, '12'),
+])
```

Comment:
I'd suggest adding a test that checks the not applying corrections twice behaviour.

Something like:

{"abc 123": "456", "456": "789"}
"abc 123 456"
becomes "456 789"

A test case that demonstrates the order-dependence would be useful too, both of strings and regexes

Also a test for the regex running after the words behaviour.